opm/output/OutputWriter.hpp

```diff
@@ -74,44 +75,28 @@ class OutputWriter {
      * This routine should be called before the first timestep (i.e. when
      * timer.currentStepNum () == 0)
      */
-    virtual void writeInit(const SimulatorTimerInterface &timer, const NNC& nnc = NNC()) = 0;
+    virtual void writeInit( const NNC& nnc = NNC() ) = 0;
 
     /*!
      * \brief Write a blackoil reservoir state to disk for later inspection with
      *        visualization tools like ResInsight
      *
-     * \param[in] timer          The timer providing time, time step, etc. information
-     * \param[in] reservoirState The thermodynamic state of the reservoir
-     * \param[in] wellState      The production/injection data for all wells
+     * \param[in] report_step           The current report step
+     * \param[in] current_posix_time    Seconds elapsed since epoch
+     * \param[in] seconds_elapsed       Seconds elapsed since simulation start
+     * \param[in] reservoirState        The thermodynamic state of the reservoir
+     * \param[in] wells                 Well data
      *
      * This routine should be called after the timestep has been advanced,
      * i.e. timer.currentStepNum () > 0.
      */
-    virtual void writeTimeStep(const SimulatorTimerInterface& timer,
-                               const SimulationDataContainer& reservoirState,
-                               const WellState& wellState,
-                               bool  isSubstep) = 0;
+    virtual void writeTimeStep( int report_step,
+                                time_t current_posix_time,
+                                double seconds_elapsed,
+                                data::Solution reservoirState,
+                                data::Wells,
+                                bool  isSubstep) = 0;
```

Comment:
I totally agree and it's a much better interface.

I'll make a note of it and add it to the list of interface improvements (the isSubstep logic is inherited from the old implementation).